File: snapshots/output-net8.0/syntax/VBMain/Classes.vb
  Imports System.Diagnostics.CodeAnalysis
'         ^^^^^^ reference scip-dotnet nuget . . System/
'                ^^^^^^^^^^^ reference scip-dotnet nuget . . Diagnostics/
'                            ^^^^^^^^^^^^ reference scip-dotnet nuget . . CodeAnalysis/

  Namespace VBMain
'           ^^^^^^ reference scip-dotnet nuget . . VBMain/
      <SuppressMessage("ReSharper", "all")>
'      ^^^^^^^^^^^^^^^ reference scip-dotnet nuget System.Runtime 7.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
      Public Class Classes
'                  ^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#
'                          documentation ```vb\nClass Classes\n```

          Public Name As String
'                ^^^^ definition scip-dotnet nuget . . VBMain/Classes#Name.
'                     documentation ```vb\nPublic Classes.Name As String\n```
          Public Const IntConstant As Integer = 1
'                      ^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#IntConstant.
'                                  documentation ```vb\nPublic Const Classes.IntConstant As Integer = 1\n```
          Public Const StringConstant As String = "hello"
'                      ^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#StringConstant.
'                                     documentation ```vb\nPublic Const Classes.StringConstant As String = "hello"\n```

          Public Sub New(ByVal name As Integer)
'                    ^^^ definition scip-dotnet nuget . . VBMain/Classes#`.ctor`().
'                        documentation ```vb\nPublic Sub Classes.New(name As Integer)\n```
'                              ^^^^ definition scip-dotnet nuget . . VBMain/Classes#`.ctor`().(name)
'                                   documentation ```vb\nname As Integer\n```
              Me.Name = "name"
'                ^^^^ reference scip-dotnet nuget . . VBMain/Classes#Name.
          End Sub

          Public Sub New(ByVal name As String)
'                    ^^^ definition scip-dotnet nuget . . VBMain/Classes#`.ctor`(+1).
'                        documentation ```vb\nPublic Sub Classes.New(name As String)\n```
'                              ^^^^ definition scip-dotnet nuget . . VBMain/Classes#`.ctor`(+1).(name)
'                                   documentation ```vb\nname As String\n```
              Me.Name = name
'                ^^^^ reference scip-dotnet nuget . . VBMain/Classes#Name.
'                       ^^^^ reference scip-dotnet nuget . . VBMain/Classes#`.ctor`(+1).(name)
          End Sub

          Protected Overrides Sub Finalize()
'                                 ^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#Finalize().
'                                          documentation ```vb\nProtected Overrides Sub Classes.Finalize()\n```
'                                          relationship implementation reference scip-dotnet nuget System.Runtime 7.0.0.0 System/Object#Finalize().
              Console.WriteLine(42)
'             ^^^^^^^ reference scip-dotnet nuget System.Console 7.0.0.0 System/Console#
'                     ^^^^^^^^^ reference scip-dotnet nuget System.Console 7.0.0.0 System/Console#WriteLine(+7).
          End Sub

          Public Class ObjectClass
'                      ^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#ObjectClass#
'                                  documentation ```vb\nClass ObjectClass\n```
'                                  relationship implementation scip-dotnet nuget . . VBMain/Classes#SomeInterface#
              Inherits Object
              Implements SomeInterface
'                        ^^^^^^^^^^^^^ reference scip-dotnet nuget . . VBMain/Classes#SomeInterface#
          End Class

          Public Partial Class PartialClass
'                              ^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#PartialClass#
'                                           documentation ```vb\nClass PartialClass\n```
          End Class

          Class TypeParameterClass(Of T)
'               ^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#TypeParameterClass#
'                                  documentation ```vb\nClass TypeParameterClass(Of T)\n```
'                                     ^ definition local 0
'                                       documentation ```vb\nT\n```
          End Class

          Friend Class InternalMultipleTypeParametersClass(Of T1, T2)
'                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#InternalMultipleTypeParametersClass#
'                                                          documentation ```vb\nClass InternalMultipleTypeParametersClass(Of T1, T2)\n```
'                                                             ^^ definition local 1
'                                                                documentation ```vb\nT1\n```
'                                                                 ^^ definition local 2
'                                                                    documentation ```vb\nT2\n```
          End Class

          Interface ICovariantContravariant(Of In T1, Out T2)
'                   ^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#ICovariantContravariant#
'                                           documentation ```vb\nInterface ICovariantContravariant(Of In T1, Out T2)\n```
'                                                 ^^ definition local 3
'                                                    documentation ```vb\nIn T1\n```
'                                                         ^^ definition local 4
'                                                            documentation ```vb\nOut T2\n```
              Sub Method1(ByVal t1 As T1)
'                 ^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#ICovariantContravariant#Method1().
'                         documentation ```vb\nSub ICovariantContravariant(Of In T1, Out T2).Method1(t1 As T1)\n```
'                               ^^ definition scip-dotnet nuget . . VBMain/Classes#ICovariantContravariant#Method1().(t1)
'                                  documentation ```vb\nt1 As T1\n```
'                                     ^^ reference local 3

              Function Method2() As T2
'                      ^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#ICovariantContravariant#Method2().
'                              documentation ```vb\nFunction ICovariantContravariant(Of In T1, Out T2).Method2() As T2\n```
'                                   ^^ reference local 4

          End Interface

          Public Class StructConstraintClass(Of T As Structure)
'                      ^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#StructConstraintClass#
'                                            documentation ```vb\nClass StructConstraintClass(Of T As Structure)\n```
'                                               ^ definition local 5
'                                                 documentation ```vb\nT\n```
          End Class

          Public Class ClassConstraintClass(Of T As Class)
'                      ^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#ClassConstraintClass#
'                                           documentation ```vb\nClass ClassConstraintClass(Of T As Class)\n```
'                                              ^ definition local 6
'                                                documentation ```vb\nT\n```
          End Class

          Public Class NewConstraintClass(Of T As New)
'                      ^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#NewConstraintClass#
'                                         documentation ```vb\nClass NewConstraintClass(Of T As New)\n```
'                                            ^ definition local 7
'                                              documentation ```vb\nT\n```
          End Class

          Public Class TypeParameterConstraintClass(Of T As SomeInterface)
'                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#TypeParameterConstraintClass#
'                                                   documentation ```vb\nClass TypeParameterConstraintClass(Of T As SomeInterface)\n```
'                                                      ^ definition local 8
'                                                        documentation ```vb\nT\n```
'                                                           ^^^^^^^^^^^^^ reference scip-dotnet nuget . . VBMain/Classes#SomeInterface#
          End Class

          Private Class MultipleTypeParameterConstraintsClass(Of T1 As {SomeInterface, SomeInterface2, New}, T2 As SomeInterface2)
'                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#MultipleTypeParameterConstraintsClass#
'                                                             documentation ```vb\nClass MultipleTypeParameterConstraintsClass(Of T1 As {SomeInterface, SomeInterface2, New}, T2 As SomeInterface2)\n```
'                                                                ^^ definition local 9
'                                                                   documentation ```vb\nT1\n```
'                                                                       ^^^^^^^^^^^^^ reference scip-dotnet nuget . . VBMain/Classes#SomeInterface#
'                                                                                      ^^^^^^^^^^^^^^ reference scip-dotnet nuget . . VBMain/Classes#SomeInterface2#
'                                                                                                            ^^ definition local 10
'                                                                                                               documentation ```vb\nT2\n```
'                                                                                                                  ^^^^^^^^^^^^^^ reference scip-dotnet nuget . . VBMain/Classes#SomeInterface2#
          End Class

          Class IndexClass
'               ^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#
'                          documentation ```vb\nClass IndexClass\n```
              Private a As Boolean
'                     ^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#a.
'                       documentation ```vb\nPrivate IndexClass.a As Boolean\n```

              Default Public Property Item(ByVal index As Integer) As Boolean
'                                     ^^^^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#Item.
'                                          documentation ```vb\nPublic Default Property IndexClass.Item(index As Integer) As Boolean\n```
'                                                ^^^^^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#Item.(index)
'                                                      documentation ```vb\nindex As Integer\n```
                  Get
                      Return a
'                            ^ reference scip-dotnet nuget . . VBMain/Classes#IndexClass#a.
                  End Get
                  Set(ByVal value As Boolean)
'                           ^^^^^ definition scip-dotnet nuget . . VBMain/Classes#IndexClass#set_Item().(value)
'                                 documentation ```vb\nvalue As Boolean\n```
                      a = value
'                     ^ reference scip-dotnet nuget . . VBMain/Classes#IndexClass#a.
'                         ^^^^^ reference scip-dotnet nuget . . VBMain/Classes#IndexClass#set_Item().(value)
                  End Set
              End Property
          End Class

          Interface SomeInterface
'                   ^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#SomeInterface#
'                                 documentation ```vb\nInterface SomeInterface\n```
          End Interface

          Friend Interface SomeInterface2
'                          ^^^^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Classes#SomeInterface2#
'                                         documentation ```vb\nInterface SomeInterface2\n```
          End Interface

      End Class

  End Namespace

File: snapshots/output-net9.0/syntax/Main/Classes.cs
  using System.Diagnostics.CodeAnalysis;
//      ^^^^^^ reference scip-dotnet nuget . . System/
//             ^^^^^^^^^^^ reference scip-dotnet nuget . . Diagnostics/
//                         ^^^^^^^^^^^^ reference scip-dotnet nuget . . CodeAnalysis/

  namespace Main;
//          ^^^^ reference scip-dotnet nuget . . Main/

  [SuppressMessage("ReSharper", "all")]
// ^^^^^^^^^^^^^^^ reference scip-dotnet nuget System.Runtime 9.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
  public class Classes
//             ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#
//                     documentation ```cs\nclass Classes\n```
  {
      public string Name;
//                  ^^^^ definition scip-dotnet nuget . . Main/Classes#Name.
//                       documentation ```cs\npublic string Classes.Name\n```
      public const int IntConstant = 1;
//                     ^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#IntConstant.
//                                 documentation ```cs\npublic const int Classes.IntConstant = 1\n```
      public const string StringConstant = $"hello";
//                        ^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#StringConstant.
//                                       documentation ```cs\npublic const string Classes.StringConstant = "hello"\n```

      public Classes(int name)
//           ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#`.ctor`().
//                   documentation ```cs\npublic Classes.Classes(int name)\n```
//                       ^^^^ definition scip-dotnet nuget . . Main/Classes#`.ctor`().(name)
//                            documentation ```cs\nint name\n```
      {
          Name = "name";
//        ^^^^ reference scip-dotnet nuget . . Main/Classes#Name.
      }

      public Classes(string name) => Name = name;
//           ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#`.ctor`(+1).
//                   documentation ```cs\npublic Classes.Classes(string name)\n```
//                          ^^^^ definition scip-dotnet nuget . . Main/Classes#`.ctor`(+1).(name)
//                               documentation ```cs\nstring name\n```
//                                   ^^^^ reference scip-dotnet nuget . . Main/Classes#Name.
//                                          ^^^^ reference scip-dotnet nuget . . Main/Classes#`.ctor`(+1).(name)

      ~Classes()
//     ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#Finalize().
//             documentation ```cs\nprotected Classes.~Classes()\n```
      {
          Console.WriteLine(42);
//        ^^^^^^^ reference scip-dotnet nuget System.Console 9.0.0.0 System/Console#
//                ^^^^^^^^^ reference scip-dotnet nuget System.Console 9.0.0.0 System/Console#WriteLine(+7).
      }

      public class ObjectClass : object, SomeInterface
//                 ^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ObjectClass#
//                             documentation ```cs\nclass ObjectClass\n```
//                             relationship implementation scip-dotnet nuget . . Main/SomeInterface#
//                                       ^^^^^^^^^^^^^ reference scip-dotnet nuget . . Main/SomeInterface#
      {
      }

      public partial class PartialClass
//                         ^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#PartialClass#
//                                      documentation ```cs\nclass PartialClass\n```
      {
      }

      class TypeParameterClass<T>
//          ^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#TypeParameterClass#
//                             documentation ```cs\nclass TypeParameterClass<T>\n```
//                             ^ definition local 0
//                               documentation ```cs\nT\n```
      {
      }

      internal class InternalMultipleTypeParametersClass<T1, T2>
//                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#InternalMultipleTypeParametersClass#
//                                                       documentation ```cs\nclass InternalMultipleTypeParametersClass<T1, T2>\n```
//                                                       ^^ definition local 1
//                                                          documentation ```cs\nT1\n```
//                                                           ^^ definition local 2
//                                                              documentation ```cs\nT2\n```
      {
      }

      interface ICovariantContravariant<in T1, out T2>
//              ^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ICovariantContravariant#
//                                      documentation ```cs\ninterface ICovariantContravariant<in T1, out T2>\n```
//                                         ^^ definition local 3
//                                            documentation ```cs\nin T1\n```
//                                                 ^^ definition local 4
//                                                    documentation ```cs\nout T2\n```
      {
          public void Method1(T1 t1)
//                    ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ICovariantContravariant#Method1().
//                            documentation ```cs\nvoid ICovariantContravariant<in T1, out T2>.Method1(T1 t1)\n```
//                            ^^ reference local 3
//                               ^^ definition scip-dotnet nuget . . Main/Classes#ICovariantContravariant#Method1().(t1)
//                                  documentation ```cs\nT1 t1\n```
          {
              Console.WriteLine(t1);
//            ^^^^^^^ reference scip-dotnet nuget System.Console 9.0.0.0 System/Console#
//                    ^^^^^^^^^ reference scip-dotnet nuget System.Console 9.0.0.0 System/Console#WriteLine(+9).
//                              ^^ reference scip-dotnet nuget . . Main/Classes#ICovariantContravariant#Method1().(t1)
          }

          public T2? Method2()
//               ^^ reference local 4
//                   ^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ICovariantContravariant#Method2().
//                           documentation ```cs\nT2? ICovariantContravariant<in T1, out T2>.Method2()\n```
          {
              return default(T2);
//                           ^^ reference local 4
          }
      }

      public class StructConstraintClass<T> where T : struct
//                 ^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#StructConstraintClass#
//                                       documentation ```cs\nclass StructConstraintClass<T> where T : struct\n```
//                                       ^ definition local 5
//                                         documentation ```cs\nT\n```
//                                                ^ reference local 5
      {
      }

      public class UnmanagedConstraintClass<T> where T : unmanaged
//                 ^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#UnmanagedConstraintClass#
//                                          documentation ```cs\nclass UnmanagedConstraintClass<T> where T : unmanaged\n```
//                                          ^ definition local 6
//                                            documentation ```cs\nT\n```
//                                                   ^ reference local 6
      {
      }

      public class ClassConstraintClass<T> where T : class
//                 ^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#ClassConstraintClass#
//                                      documentation ```cs\nclass ClassConstraintClass<T> where T : class\n```
//                                      ^ definition local 7
//                                        documentation ```cs\nT\n```
//                                               ^ reference local 7
      {
      }

      public class NonNullableConstraintClass<T> where T : notnull
//                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#NonNullableConstraintClass#
//                                            documentation ```cs\nclass NonNullableConstraintClass<T> where T : notnull\n```
//                                            ^ definition local 8
//                                              documentation ```cs\nT\n```
//                                                     ^ reference local 8
      {
      }

      public class NewConstraintClass<T> where T : new()
//                 ^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#NewConstraintClass#
//                                    documentation ```cs\nclass NewConstraintClass<T> where T : new()\n```
//                                    ^ definition local 9
//                                      documentation ```cs\nT\n```
//                                             ^ reference local 9
      {
      }

      public class TypeParameterConstraintClass<T> where T : SomeInterface
//                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#TypeParameterConstraintClass#
//                                              documentation ```cs\nclass TypeParameterConstraintClass<T> where T : SomeInterface\n```
//                                              ^ definition local 10
//                                                documentation ```cs\nT\n```
//                                                       ^ reference local 10
//                                                           ^^^^^^^^^^^^^ reference scip-dotnet nuget . . Main/SomeInterface#
      {
      }

      private class MultipleTypeParameterConstraintsClass<T1, T2> where T1 : SomeInterface, SomeInterface2, new()
//                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#MultipleTypeParameterConstraintsClass#
//                                                        documentation ```cs\nclass MultipleTypeParameterConstraintsClass<T1, T2> where T1 : SomeInterface, SomeInterface2, new() where T2 : SomeInterface2\n```
//                                                        ^^ definition local 11
//                                                           documentation ```cs\nT1\n```
//                                                            ^^ definition local 12
//                                                               documentation ```cs\nT2\n```
//                                                                      ^^ reference local 11
//                                                                           ^^^^^^^^^^^^^ reference scip-dotnet nuget . . Main/SomeInterface#
//                                                                                          ^^^^^^^^^^^^^^ reference scip-dotnet nuget . . Main/SomeInterface2#
          where T2 : SomeInterface2
//              ^^ reference local 12
//                   ^^^^^^^^^^^^^^ reference scip-dotnet nuget . . Main/SomeInterface2#
      {
      }

      class IndexClass
//          ^^^^^^^^^^ definition scip-dotnet nuget . . Main/Classes#IndexClass#
//                     documentation ```cs\nclass IndexClass\n```
      {
          private bool a;
//                     ^ definition scip-dotnet nuget . . Main/Classes#IndexClass#a.
//                       documentation ```cs\nprivate bool IndexClass.a\n```

          public bool this[int index]
//                             ^^^^^ definition scip-dotnet nuget . . Main/Classes#IndexClass#`this[]`.(index)
//                                   documentation ```cs\nint index\n```
          {
              get { return a; }
//                         ^ reference scip-dotnet nuget . . Main/Classes#IndexClass#a.
              set { a = value; }
//                  ^ reference scip-dotnet nuget . . Main/Classes#IndexClass#a.
//                      ^^^^^ reference scip-dotnet nuget . . Main/Classes#IndexClass#set_Item().(value)
          }
      }

  }

  public interface SomeInterface
//                 ^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/SomeInterface#
//                               documentation ```cs\ninterface SomeInterface\n```
  {
  }

  internal interface SomeInterface2
//                   ^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/SomeInterface2#
//                                  documentation ```cs\ninterface SomeInterface2\n```
  {
  }
